test/python/test_schemas.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=redefined-builtin
+
+""" Schemas test"""
+import json
+import os
+from qiskit import (validate_json_against_schema, _create_validator)
+from qiskit import __path__ as qiskit_path
+from .common import QiskitTestCase
+
+
+class TestSchemaExamples(QiskitTestCase):
+    """
+    Tests schema validation
+    """
+
+    def setUp(self):
+        self.schema_tests = []
+        self.schema_tests.append(
+            {"schema_name": "backend_configuration",
+             "examples": [
+                 "backend_configuration_openpulse_example.json",
+                 "backend_configuration_openqasm_example.json",
+                 "backend_configuration_openqasm_simulator_example.json"]})
+        self.schema_tests.append(
+            {"schema_name": "backend_properties",
+             "examples": [
+                 "backend_properties_example.json"]})
+        self.schema_tests.append(
+            {"schema_name": "backend_status",
+             "examples": [
+                 "backend_status_example.json"]})
+        self.schema_tests.append(
+            {"schema_name": "default_pulse_configuration",
+             "examples": [
+                 "default_pulse_configuration_example.json"]})
+        self.schema_tests.append(
+            {"schema_name": "job_status",
+             "examples": [
+                 "job_status_example.json"]})
+        self.schema_tests.append(
+            {"schema_name": "qobj",
+             "examples": [
+                 "qobj_openpulse_example.json",
+                 "qobj_openqasm_example.json"]})
+        self.schema_tests.append(
+            {"schema_name": "result",
+             "examples": [
+                 "result_openqasm_example.json",
+                 "result_openpulse_level_0_example.json",
+                 "result_openpulse_level_1_example.json",
+                 "result_statevector_simulator_example.json",
+                 "result_unitary_simulator_example.json"]})
+        self.examples_base_path = os.path.join(qiskit_path[0], 'schemas',
+                                               'examples')
+
+    def test_examples_are_valid(self):
+        """ Validate example json files against respective schemas"""
+        for schema_test in self.schema_tests:
+            schema_name = schema_test['schema_name']
+            with self.subTest(schema_test=schema_name):
```

Comment:
Non-blocking: the nested `subTest` does not help to the readability of the code.  Would it be too messy to separate each schema in its own method? Do we think they are growing in number? If they will remain almost the same, I would make each set of example validations explicit in its own `test_schema_name_examples_are_valid` method.